### ISSUE TITLE
Comprehensive List Functions

### ISSUE THREAD

**Issue by chriseppstein** [sass/sass#380]
Adding two lists `$a + $b` should have the semantics as `join($a, $b)`

The  `-` operator should remove elements from the first list that are found in the second list. This is great for doing things like compacting a list: `$a - (false, null)`

**Comment by Snugug** [comment]
+1 for this feature, and more robust list operators/functions in general.

**Comment by nex3** [comment]
`+` is reasonable, since that's familiar to JavaScript users and pretty intuitive. `-` is less so; Ruby is the only language I can think of that defines `-` in this way. I don't think list manipulation is important enough to Sass to make it worthwhile to add a shortcut for that operation.

In any case, this is something that I want to push off until 3.3.

**Comment by Snugug** [comment]
@nex3 I'd say that list manipulation may not be very important to every day Sass users, but for those of us building frameworks in Sass, lists are very important and using and manipulating them winds up being a fairly hacky process. More robust list, and for that matter string, functions in Sass would be a great boon to the language.

**Comment by chriseppstein** [comment]
I agree with this sentiment. But I think @nex3's point was simply that the behavior needn't be core to the language. A list function may suffice. 

Hunt & pecked on my iPhone... Sorry if it's brief!

**Comment by chriseppstein** [comment]
We should add a `list-slice($from, $to)`, `list-insert($after-index)` and some other helpful list methods.
